File: java-spring-moshi/src/main/java/test_service/services/v2/EchoServiceImpl.java
package test_service.services.v2;

import org.springframework.stereotype.Service;
import test_service.v2.models.Message;
import test_service.v2.services.echo.EchoService;

@Service("EchoServiceV2")
public class EchoServiceImpl implements EchoService {
	@Override
	public Message echoBodyModel(Message body) {
		return body;
	}
}

File: kotlin-micronaut/src/main/kotlin/test_service/services/v2/EchoServiceImpl.kt
package test_service.services.v2

import io.micronaut.context.annotation.Bean
import test_service.v2.models.Message
import test_service.v2.services.echo.EchoService

@Bean
class EchoServiceImpl : EchoService {
    override fun echoBodyModel(body: Message): Message {
        return body
    }
}